AsyncDisplayKit/ASCellNode.m

```diff
@@ -106,21 +105,12 @@ - (instancetype)init
   return self;
 }
 
-- (CGSize)calculateSizeThatFits:(CGSize)constrainedSize
+- (id<ASLayoutable>)layoutSpecThatFits:(ASSizeRange)constrainedSize
```

Comment:
If a display node want to return a ASLayout directly (kind of semi-auto layout), it can overrides -layoutThatFits: and return a flat ASLayout that describe its size, as well as position and size of subnodes. I think the new APIs added to ASDisplayNode can support a great deal of use cases, including auto, semi auto and manual layout.